src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java

```diff
@@ -28,35 +32,75 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
+                builder.append("Backwards compatibility enabled!\n").color(ChatColor.RED);
             }
 
-            sender.sendMessage("");
-
-            Collection<Plugin> addons = SlimefunPlugin.getInstalledAddons();
-            sender.sendMessage(ChatColors.color("&7Installed Addons: &8(" + addons.size() + ")"));
+            builder.append("\n");
 
-            for (Plugin plugin : addons) {
-                String version = plugin.getDescription().getVersion();
+            addPluginVersions(builder);
 
-                if (Bukkit.getPluginManager().isPluginEnabled(plugin)) {
-                    sender.sendMessage(ChatColor.GREEN + " " + plugin.getName() + ChatColor.DARK_GREEN + " v" + version);
-                } else {
-                    sender.sendMessage(ChatColor.RED + " " + plugin.getName() + ChatColor.DARK_RED + " v" + version);
-                }
-            }
+            System.out.println(ComponentSerializer.toString(builder.create()));
+            sender.spigot().sendMessage(builder.create());
         } else {
             SlimefunPlugin.getLocalization().sendMessage(sender, "messages.no-permission", true);
         }
     }
 
+    private void addJavaVersion(@Nonnull ComponentBuilder builder) {
+        String javaVer = System.getProperty("java.version");
+        if (javaVer.startsWith("1.")) {
+            javaVer = javaVer.substring(2);
+        }
+        if (javaVer.indexOf('.') != -1) { // If it's like 11.0.1.3 or 8.0_275
+            javaVer = javaVer.substring(0, javaVer.indexOf('.'));
+        }
+        int ver = Integer.parseInt(javaVer);
+
+        if (ver < 11) {
+            builder.append("Java " + ver).color(ChatColor.RED)
+                .event(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new Text(
+                    "You should be using at least Java 11! Paper will be dropping support for before Java 11 starting at MC 1.17"
+                )))
+                .append("\n")
+                .event((HoverEvent) null);
+        } else {
+            builder.append("Java " + ver + "\n").color(ChatColor.GREEN);
+        }
+    }
+
+    private void addPluginVersions(@Nonnull ComponentBuilder builder) {
+        Collection<Plugin> addons = SlimefunPlugin.getInstalledAddons();
+        builder.append("Installed Addons: ").color(ChatColor.GRAY)
+            .append("(" + addons.size() + ")").color(ChatColor.DARK_GRAY);
+
+        for (Plugin plugin : addons) {
+            String version = plugin.getDescription().getVersion();
+
+            if (Bukkit.getPluginManager().isPluginEnabled(plugin)) {
+                builder.append("\n  " + plugin.getName()).color(ChatColor.GREEN)
+                    .append(" v" + version).color(ChatColor.DARK_GREEN);
+            } else {
+                HoverEvent hover = new HoverEvent(HoverEvent.Action.SHOW_TEXT,
+                    new Text("Plugin is disabled. Check the console for an error and report on their issue tracker."));
```

Comment:
Should probably do an `instanceof SlimefunAddon` check here and link the issues tracker then.